### PR TITLE
feat: configure logical replication stream with our display settings

### PR DESCRIPTION
**Pull request by kevin-dp**
This PR configures the logical replication stream to use well-defined display settings for formatting the value that come on the replication stream.

### CODE REVIEW

**Review by alco — COMMENTED** [review]
Looks great!

[`IntervalStyle`](https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-INTERVALSTYLE) is missing from the list of display settings. The original Electric didn't setit  because it didn't support that type.

We should pick a format for intervals like we have for dates. Though I can't say off the top of my head which of these is preferable - https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-INTERVAL-INPUT.

**Comment by kevin-dp** [comment]
> Looks great!
> 
> [`IntervalStyle`](https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-INTERVALSTYLE) is missing from the list of display settings. The original Electric didn't setit because it didn't support that type.
> 
> We should pick a format for intervals like we have for dates. Though I can't say off the top of my head which of these is preferable - https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-INTERVAL-INPUT.

Right, will look into it 👍

**Comment by kevin-dp** [comment]
@alco I added `IntervalStyle`. Decided to go for `iso_8601` because it is compact yet explicit and can be parsed by libraries such as moment.js. Downside is it is not part of the SQL standard so it loses compatibility with other DBs. We can change it to `sql_standard` if we want to prioritise DB compatibility.

File: packages/sync-service/lib/electric/postgres.ex [hunk 1]
```diff
@@ -32,4 +32,34 @@ defmodule Electric.Postgres do
...
+
+    * `bytea_output`       - determines how Postgres encodes bytea values. It can use either Hex- or
+                             Escape-based encoding.
+
+    * `DateStyle`          - determines how Postgres interprets date values.
+
+    * `TimeZone`           - affects the time zone offset Postgres uses for timestamptz and timetz values.
+
+    * `extra_float_digits` - determines whether floating-point values are rounded or are encoded precisely.
+    * `IntervalStyle`      - determines how Postgres interprets and formats interval values.
```
Comment (alco) [inline packages/sync-service/lib/electric/postgres.ex:61]:
```suggestion

    * `IntervalStyle`      - determines how Postgres interprets and formats interval values.
```

**Review by alco — APPROVED** [review]
Nice!

I'm a bit on the fence about IntervalStyle. ISO is a safe choice and consistent with the other settings, but when I work with intervals in `psql`, I prefer the default postgres format.

@icehaunter has already implemented parser for all interval formats in Elixir, so maybe we could support them all? But then the argument of restricting each multi-format type to a single format for simplicity and consistency goes out the window. Sigh...

This is good to merge. I may yet bring up the topic of supported formats in `where` clauses at an upcoming team standup to gather opinions.

**Comment by kevin-dp** [comment]
> Nice!
> 
> I'm a bit on the fence about IntervalStyle. ISO is a safe choice and consistent with the other settings, but when I work with intervals in `psql`, I prefer the default postgres format.
> 
> @icehaunter has already implemented parser for all interval formats in Elixir, so maybe we could support them all? But then the argument of restricting each multi-format type to a single format for simplicity and consistency goes out the window. Sigh...
> 
> This is good to merge. I may yet bring up the topic of supported formats in `where` clauses at an upcoming team standup to gather opinions.

I think it makes sense to stick to a single format when serving shape data.
For `where` clauses it makes sense to support the different styles if we already have a parser for it anyway. As you suggested, let's merge this and see if we can add support for the different input styles for `where` clauses in a separate PR.

**Pull request merged**